docs/sorting-algorithms/selection-sort.mdx

```diff
@@ -0,0 +1,206 @@
+---
+title: Selection Sort
+slug: /sorting-algorithms/selection-sort
+---
+
+# Introduction
+
+Selection Sort is a sorting algorithm which finds out the smallest element in unsorted part in each iteration and put it to the last of sorted part.
+It does less memory writes as compared to other sorting algorithms.
+
+# Algorithm
+
+In Selection Sort, we follow the following algorithm:
+
+* First we will find the smallest element in the range [0,n-1] and swap it with the element at 0th index
+
+* Then we will find the smallest element int the range [1,n-1] and swap it with the element at 1st index
+
+* We keep on increasing the left part of the range until it reaches n-1 and do the above mentioned process
+
+# Explanation for Selection Sort
+
+### Example 1
+
+```
+   0   1   2   3
+ | 4 | 1 | 2 | 7 |
+
+ [0,3] min_idx=1 
+ swap(arr[min_idx],arr[0]);
+
+ | 1 | 4 | 2 | 7 |
+
+ [1,3] min_idx=2
+ swap(arr[min_idx],arr[1]);
+
+ | 1 | 2 | 4 | 7 |
+
+ [2,3] min_idx=2 yar 
+ swap(arr[min_idx],arr[2]);
+```
+### Example 2
+
+```
+   0   1   2   3
+ | 1 | 8 | 5 | 2 |
+
+ [0,3] min_idx=0
+ swap(arr[min_idx],arr[0]);
+
+ | 1 | 8 | 5 | 2 |
+
+ [1,3] min_idx=3
+ swap(arr[min_idx],arr[1]);
+
+ | 1 | 2 | 4 | 8 |
+
+ [2,3] min_idx=2 
+ swap(arr[min_idx],arr[2]);
+```
```

Comment:
Can you use a more intuitive example, please?

Reply:
done